project_task_code/models/project_task.py

```diff
@@ -28,3 +29,36 @@ def copy(self, default=None):
             default = {}
         default['code'] = self.env['ir.sequence'].next_by_code('project.task')
         return super(ProjectTask, self).copy(default)
+
+    @api.model
+    def name_search(self, name='', args=None, operator='ilike', limit=100):
+        args = args or []
+        code = self._name_search_parse_code(name)
+        if code:
+            # We are searching for codes
+            tasks = self.search(
+                [('code', '=ilike', code + '%')] + args,
+                limit=limit)
+            if tasks:
+                return tasks.name_get()
+        return super(ProjectTask, self).name_search(
+            name=name,
+            args=args,
+            operator=operator,
+            limit=limit)
+
+    def _name_search_parse_code(self, name):
+        # Remove any whitespace from name
+        name = name.lstrip()
+        # Get the data record
+        sequence_id = self.env.ref('project_task_code.sequence_task')
+        prefix = sequence_id.prefix or ''
+        suffix = sequence_id.suffix or ''
+        # Check if it looks like code
+        if not all([
+            name.startswith(prefix),
+            name.endswith(suffix),
+            name.lstrip(prefix + suffix).isdigit(),
```

Comment:
consider

```
>>> 'abc42bca'.lstrip('cab')
'42bca'
```

vs

```
>>> 'abc42bca'[len('abc'):-len('bca')]
'42'
```

The corner case is when the length of `suffix` is 0, for that you'll need to add an `or len(name)`

Reply:
I do not understand your last sentence.

Reply:
```python
>>> prefix = 'pre'
>>> suffix = 'suffix'
>>> name = 'pre4242suffix'
>>> name[len(prefix):-len(suffix)]
'4242'
>>> suffix = ''
>>> name = 'pre4242'
>>> name[len(prefix):-len(suffix)]
''
>>> name[len(prefix):-len(suffix) or len(name)]
'4242'
```